src/net.h

```diff
@@ -1486,12 +1486,8 @@ class CNode
 
     void PushInventory(const CInv& inv)
     {
-        if (inv.type == MSG_BLOCK) {
-            LogPrint(BCLog::NET, "%s -- adding new inv: %s peer=%d\n", __func__, inv.ToString(), id);
-            LOCK(cs_inventory);
-            vInventoryBlockToSend.push_back(inv.hash);
-            return;
-        }
+        assert(inv.type != MSG_BLOCK);
+
```

Comment:
19347: why did this change?

Reply:
> 19347: why did this change?

because `PushInventory` not meant to push any blocks; blocks are supposed to be processed by its own code:
 - `pfrom.vInventoryBlockToSend` - different storage are used for blocks intentionally
 - critical sections are under different mutexes also: `pnode->cs_inventory` vs `m_tx_relay->cs_tx_inventory` 

There's further related refactorings in other backports such as bitcoin#19610 which split block/other inventory processing futher

Reply:
see also related Udjin's comment to this PR https://github.com/dashpay/dash/pull/5825#discussion_r1455748480

Reply:
Well, maybe just a tiny bit 😄  It would be a logical error if we'd use it for blocks so it should never happen/should crash in tests. But we can do it in a slightly safer way I guess... smth like
```suggestion
        ASSERT_IF_DEBUG(inv.type != MSG_BLOCK);
        if (inv.type == MSG_BLOCK) {
            LogPrintf("%s -- WARNING: using PushInventory for BLOCK inv, peer=%d\n", __func__, id);
            return;
        }
```
maybe?

Reply:
@PastaPastaPasta I am ok with current version, if there's a block on PushInventory - something went wrong not due to misbehaving network peers, but due to programmer error...

up to you if I need to change it if you can think a situation that this assert will shoot